mule-extensions-api/src/main/java/org/mule/runtime/extension/api/client/OperationParameterizer.java

```diff
@@ -0,0 +1,165 @@
+
+  /**
+   * Specifies that the operation is to used whatever repeatable iterables strategy is default for the current execution context.
+   * <p>
+   * If this concept does not apply to the executed operation (either because it uses bytes streaming instead of iterables or
+   * because it doesn't stream at all), then calling this method has no effect.
+   * <p>
+   * If no repeatable streaming strategy is specified (either through this or similar methods), then streaming results will not be
+   * made repeatable
+   *
+   * @return {@code this} instance
+   */
+  OperationParameterizer withDefaultRepeatableIterables();
+
+  /**
+   * Specifies that the operation is to use in-memory repeatable iterables
+   * <p>
+   * If this concept does not apply to the executed operation (either because it uses bytes streams instead of iterables or
+   * because it doesn't stream at all), then calling this method has no effect.
+   * <p>
+   * If no repeatable streaming strategy is specified (either through this or similar methods), then streaming results will not be
+   * made repeatable
+   *
+   * @param initialBufferSize   the buffer's initial size. Must be greater than zero
+   * @param bufferSizeIncrement the size that the buffer should gain each time it is expanded. A value of zero means no expansion.
+   *                            Cannot be lower than zero.
+   * @param maxBufferSize       the maximum amount of space that the buffer can grow to. Use {@code null} for unbounded buffers
+   * @return {@code this} instance
+   */
+  OperationParameterizer withInMemoryRepeatableIterables(int initialBufferSize, int bufferSizeIncrement, int maxBufferSize);
+
+  /**
+   * Specifies that the operation is to use file based repeatable streaming
+   * <p>
+   * If this concept does not apply to the executed operation (either because it uses bytes streams instead of iterables or
+   * because it doesn't stream at all), then calling this method has no effect.
+   * <p>
+   * If no repeatable streaming strategy is specified (either through this or similar methods), then streaming results will not be
+   * made repeatable
+   *
+   * @param maxInMemoryInstances maxInMemoryInstances the maximum amount of space that the buffer can grow to. Use {@code null}
+   *                             for unbounded buffers
+   * @return {@code this} instance
+   */
+  OperationParameterizer withFileStoreRepeatableIterables(int maxInMemoryInstances);
+
+  /**
+   * Specifies an event to which this operation is relative to. This means that parameters assigned with expression values, or
+   * managed streams returned will be associated to this event. If not specified, a temporary event will be generated for the
+   * execution and discarded immediately after.
+   *
+   * @param event an {@link Event}
+   * @return {@code this} instance
+   */
+  OperationParameterizer inTheContextOf(Event event);
```

Comment:
This is a method that the Java SDK developer should never use right?

Reply:
Java SDK users can't use it even if they wanted to. Priviledged extensions such as ApiKit might use it. That been said, the primary use case for this client is not extensions but Mule itself.